src/constraint/encoding.ts

```diff
@@ -162,7 +162,7 @@ export const ENCODING_CONSTRAINTS: EncodingConstraintModel[] = [
     strict: false,
     satisfy: (encQ: EncodingQuery, schema: Schema, opt: QueryConfig) => {
       if (encQ.timeUnit && encQ.type === Type.TEMPORAL) {
-        return schema.cardinality(encQ, false, true) > 1;
+        return schema.timeUnitHasVariation(encQ);
```

Comment:
to make it consistent with the method, rename constraint to `timeUnitShouldHaveVariation`?